### ISSUE TITLE
boincmgr.exe --namehost or /n does not work on Windows

### ISSUE THREAD

**Issue by txcanyon** [BOINC/boinc#2739]
I have not been able to get the --namehost or /n flags to work on my computer running a VM and trying to connect to the boinc running inside the VM

Connecting through the Manager gui "File -> Select Computer" works fine

if I attempt to connect to the VM from my Linux Mint machine using the correct syntax for linux: 
 boincmgr -n 192.168.2.22, it does connect properly

Neither the current version 7.12.1 nor the previous 7.10.2 work

my command line is pretty simple: C:\Archives\Boinc\boincmgr.exe /n 192.168.2.22 C:\Archives\Boinc\boincmgr.exe --namehost=192.168.2.22 does not work either

I have used the --verbose flag and it does not seem to work either as none of the err files are showing anything unless they are meant for the client not the manager, but I get no errors at all on the screen using either command-line. However if I type the flag wrong, it will give me a popup and tell me the argument

**Comment by AenBleidd** [comment]
Was not able to reproduce this local debug version (7.14.1) and remote boinc client running on RasperryPi:
![image](https://user-images.githubusercontent.com/3234209/46587496-04d87580-ca96-11e8-8efa-0b5fbabcf5aa.png)

**Comment by AenBleidd** [comment]
What OS do you have inside VM?
Are you sure you pass the password correct?

**Comment by RichardHaselgrove** [comment]
I've reproduced this across a LAN (Windows --> Windows). No error messages found in either the BOINC data directory or AppData.

**Comment by AenBleidd** [comment]
Ok, in Debug mode it works fine. I'll try to reproduce this in Release mode later this evening

**Comment by AenBleidd** [comment]
It makes no sense to use /nd flag because it affects local client only. If address and password are specified boincmgr should not try to start local client. This behavior should be verified also.

**Comment by txcanyon** [comment]
Adding /nd has no effect on outcome. 
To reiterate the setup I am trying to get to work:
Host OS: Windows 10; IP=192.168.2.2
Guest OS: Ubuntu 18.04.1 inside Hyper-V with dedicated Intel 1000PT network card; IP 192.168.2.22

I do not have a GUI password set, but for giggles I manually set a simple 5 letter password, restarted the boinc-client inside the VM and reconnected manually with password and success as expected. When using the /p or --password=<password> on a command line it does not connect and as previously reported there is no log to view on my end. Shouldn't there be to diagnose issues such as these?

**Comment by AenBleidd** [comment]
I tried next scenario with 7.12.1 on both machines:
Environment 1: Windows 7, GUI RPC on, password set
Environment 2: Windows 10.
Both environments are separate physical machines within the same LAN
From Environment 2 I've successfully connected to environment 1 setting namehost and password.

@RichardHaselgrove, could you please share your steps to reproduce this?

Thanks

**Comment by txcanyon** [comment]
Ok, seemed to have narrowed down something.

If I install to the default suggested directory "\Program Files\Boinc" it WILL connect automatically.
If I install to something else ie C:\Archives\Boinc it WILL NOT connect automatically

**Comment by RichardHaselgrove** [comment]
I was trying with two Windows 7 machines, both on same LAN with local 192.168.x.y addressing. Using your numbering, 

Environment 1 is downstairs, logged in, BOINC started from Manager, Manager minimised, monitor off.

Environment 2 is my daily monitoring machine, runs BoincView (so I can see the status of Env1 at all times), also runs BOINC client autostarted by Manager.

Both machines have the same GUI RPC password set - which may become relevant later.

For testing, I exited the manager on Env2, leaving the client running. I opened a command window at the BOINC programs directory, and tested various permutations of

`boincmgr /n hostname /p password /nd`
(including replacing hostname with ipaddress)

Consistently, **NO** connection was made to the remote (Env1) machine when (and only when) /nd was on the command line.

Connection was made with /n hostname /p password (as it should), and also - to my surprise - with just /n hostname. That should have caused an error, but maybe the use of the same password on both machines avoided that - the running client or registry entry for the data directory may have supplied the necessary access to the local password.

Both machines are set to display the exit dialog. Manager windows with remote connections in general do not show the exit dialog, but in this test there was one exception: with /n hostname /nd, the exit dialog appeared when exiting the Manager for the next test.

**Comment by RichardHaselgrove** [comment]
@txcanyon 

> If I install...

That sounds like your program location. Did you test any changes to the data directory location?

> W10 -> W10 = fail
W10 -> Linux (VM) = fail
W10 -> Linux = fail

That sounds as if the second W10 machine isn't authorised as a 'remote manager' in the remote_hosts.cfg file for the clients you're trying to reach.

**Comment by txcanyon** [comment]
It is authorized because if I manually connect it works fine. The /n flag is the hangup, I can always connect manually via File -> Select Computer

**Comment by AenBleidd** [comment]
@RichardHaselgrove, pr #2745 should fix your case. Please, review it

**Comment by AenBleidd** [comment]
Ok, tried to reproduce this.
Me environment:
Host machine: Windows 7 x64, BOINC Manager 7.14.1
Guest machine on VirtualBox: Ubuntu 16.04, BOINC client 7.14.1 (don't look at the version in BOINC Manager, BOINC client was build from master branch)
BOINC Managet started from command line with namehost and password specified.
Everything works fine.
![image](https://user-images.githubusercontent.com/3234209/46815636-f46d1700-cd83-11e8-874a-b29ae8afaff6.png)

**Comment by txcanyon** [comment]
Just tried a brand new install of Win10 build 1607. Boinc 7.14.2

Installed Boinc to default directory. Tried /n 192.168.2.21 and --namehost=192.168.2.21. It does not connect
.21 is a linux system running Ubuntu 18.04.01. If I select it manually from within the Manager, it connects fine.

There must be something going on with W10 if you're having success on W7.

**Comment by AenBleidd** [comment]
Same results for Win10 (BOINC Manager 7.12.1 used)

![image](https://user-images.githubusercontent.com/3234209/46867756-c690dc80-ce2e-11e8-8808-fc5ecf8f9ad5.png)

**Comment by txcanyon** [comment]
So found a little headway here. I reinstalled W10 a 3rd time. I did NOT reinstall Boinc post W10 reinstall but used the previous install directory. I ran boincmgr.exe from \boinc_old

It connects!!! No matter what my destination was, either a VM on the same machine or another on the local network, boingmgr.exe /n 192.168.2.x /p password connected perfectly. 

So I saw the new 7.14.2 devlopment  build and installed it. Now I CANNOT connect via /n no matter the install location; it does not connect via command line
So new install does not work
and now the old install no longer works

Uninstalling Boinc does not resolve it. No matter what even running boincmgr.exe /n from the previous old directory no long works. So there is some sort of setting I'm guessing in the registry that is now set.
